### PR TITLE
fix: ensure data integrity during startup

### PR DESCRIPTION
**Pull request by tianfeng92**
## Description

Ensure records in memento exist in storage.

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -52,14 +52,26 @@ export class GlobalStorage {
...
-    // Example: $USER/Library/Application Support/Code/User/globalStorage/undefined_publisher.vscode-scriptiq/v1/scriptiq_history
+    // Example: /Users/$USER/Library/Application\ Support/Code/User/globalStorage/saucelabs.vscode-scriptiq/v1/scriptiq_history/
     return vscode.Uri.joinPath(
       this.storageUri,
       'scriptiq_history',
       ...segments,
     );
   }
 
+  verifyTestIds(ids: string[]): string[] {
```
Comment (alexplischke) [inline src/storage.ts:63]: 1. A method like `deleteTestRecord` is self explanatory and doesn't have a strong need for documentation, but `verifyTestIds` does.
2. `verifyTestIds` should be either called `verifyTestIDs` to be case-consistent with memento, or memento should be changed to follow this case instead. ⬅️ ➡️

File: src/extension.ts [hunk 2]
```diff
@@ -37,6 +37,9 @@ export async function activate(context: vscode.ExtensionContext) {
     }
     await memento.saveSchemaVersion(storage.schemaVersion);
   }
+  const testIDs = memento.getTestIDs();
```
Comment (alexplischke) [inline src/extension.ts:40]: Would be worthwhile to add a note here for our future selves, explaining why we're doing this ❔

File: src/storage.ts (hunk 1 above)
Comment (tianfeng92) [reply src/storage.ts:63]: Ah yeah. Forgot to push a commit to rename it to `verifyTestIDs` 😂

File: src/extension.ts (hunk 2 above)
Comment (tianfeng92) [reply src/extension.ts:40]: 👍

File: src/storage.ts [hunk 3]
```diff
@@ -52,14 +52,28 @@ export class GlobalStorage {
...
 
+  // Verifies the existence of test IDs by checking their corresponding URIs on the disk.
+  // Returns a filtered array containing only the IDs that have corresponding URIs.
+  verifyTestIDs(ids: string[]): string[] {
+    return ids.filter((id) => {
+      const uri = this.getHistoryUri(id);
+      try {
+        return fs.existsSync(uri.fsPath);
+      } catch (err) {
+        console.error(`Failed to check existence for ${uri.fsPath}: ${err}`);
```
Comment (alexplischke) [inline src/storage.ts:71]: "Failed to check existence" — that'd mean that this function failed to perform the file-exists check, not that the file doesn't exist ⚠️ 
```suggestion
        console.error(`Test Record ${uri.fsPath} not found: ${err}`);
```

Comment (tianfeng92) [reply src/storage.ts:71]: hmm.. If the test record doesn't exist, `fs.existsSync` will return false, not throw error. The catch block is catching unexpected err 🤔 . I can add a log to indicate the file doesn't exist.

Comment (tianfeng92) [reply src/storage.ts]: Update: `fs.existsSync` doesn't throw any error. I just remove the try catch block.

**Review by alexplischke — APPROVED** [review]

**Pull request merged**